server.js

```diff
@@ -13,6 +13,36 @@ app.get("/", function (request, response) {
   response.send("Hotel booking server.  Ask for /bookings, etc.");
 });
 
+//to make new bookings
+app.post("/bookings", (req, res) => {
+  const newBooking = req.body;
+  bookings.push(newBooking);
+  res.json(newBooking);
+});
+
+app.get("/bookings", (req, res) => res.json(bookings));
+
+// to recognise bookings by ID
+app.get("/bookings/:id", (req, res) => {
+  let id = parseInt(req.params.id);
+  const filteredBookings = bookings.find((booking) => booking.id === id);
+  res.json(filteredBookings);
+});
+
+//to delete bookings
+
+app.delete("/bookings/:id", (req, res) => {
+  let id = parseInt(req.params.id);
+  bookings.forEach((booking) => {
```

Comment:
This foreach loop means you end up trying to send back more than one response, so it crashes